### PR TITLE
[CELEBORN-520][FLINK] Tune map partition reading performance.

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
1. Refine activeMapPartitions's lock.
2.  Filter finished or released readers.

### Why are the changes needed?
To improve performance. Performance results will be added later.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
UT and cluster.

### CODE REVIEW

File: common/src/main/java/org/apache/celeborn/common/network/server/CreditStreamManager.java [hunk 1]
```diff
@@ -201,13 +222,16 @@ public void cleanResource(Long streamId) {
...
           streams.remove(streamId);
           if (mapDataPartition.getReaders().isEmpty()) {
-            synchronized (activeMapPartitions) {
-              if (mapDataPartition.getReaders().isEmpty()) {
-                mapDataPartition.close();
-                FileInfo fileInfo = mapDataPartition.getFileInfo();
-                activeMapPartitions.remove(fileInfo);
-              }
-            }
+            FileInfo fileInfo = mapDataPartition.getFileInfo();
```
[the remark this answers is not in the document]
Comment (FMX) [reply common/src/main/java/org/apache/celeborn/common/network/server/CreditStreamManager.java:225]: According to java docs, compute will remove null values mapping from hashmap.
map-compute

[the remark this answers is not in the document]
Comment (waitinfuture) [reply common/src/main/java/org/apache/celeborn/common/network/server/CreditStreamManager.java:221]: Thanks for explaining

**Comment by waitinfuture** [comment]
> only

+1, and sendDatas from the same partition reader should be in the same thread

**Review by RexXiong — APPROVED** [review]
LGTM!

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**